modules/bibcirculation/lib/bibcirculation_daemon.py

```diff
@@ -252,17 +261,94 @@ def task_run_core():
+                                      <subfield code="a">paper</subfield> \
+                                      </datafield>'
+                    if 'ebook' in type_copies or 'e-book' in type_copies:
+                        MARC_RECS_STR += '<datafield tag="340" ind1=" " ind2=" "> \
+                                          <subfield code="a">ebook</subfield> \
+                                          </datafield>'
+                lib_loc_tuples = []
+                for (_barcode, _loan_period, library_name, _library_id,
+                     location, _nb_requests,  _status, _collection,
+                     _description, _due_date) in copies:
+                    if not library_name or not location: continue
+                    if not (library_name, location) in lib_loc_tuples:
+                        lib_loc_tuples.append((library_name, location))
+                    else: continue
+                    MARC_RECS_STR += '<datafield tag="852" ind1=" " ind2=" "> \
+                                      <subfield code="c">' + library_name + '</subfield> \
+                                      <subfield code="h">' + location +'</subfield> \
+                                      </datafield>'
+                MARC_RECS_STR += '</record>'
+                recids_seen.append(recid)
+                if not done%1000:
+                    MARC_RECS_STR += "</collection>"
+                    timestamp = time.strftime("%Y-%m-%dT%H:%M:%SZ", time.localtime())
+                    marcxmlfile = 'MARCxml_booksearch' + '_' + timestamp + '_'
+                    fd, marcxmlfile = tempfile.mkstemp(dir=CFG_TMPDIR, prefix=marcxmlfile, suffix='.xml')
+                    os.write(fd, MARC_RECS_STR)
+                    os.close(fd)
+                    write_message("Composed MARCXML saved into %s" % marcxmlfile)
+ 
+                    #Schedule the bibupload task.
+                    task_id = task_low_level_submission("bibupload", "BibCirc", "-c", marcxmlfile, "-P", "3")
+                    write_message("BibUpload scheduled with task id %s" % task_id)
+                    write_message("Processed %d out of %d modified record locations." % (done, total_modified_rec_locs))
+                    MARC_RECS_STR = "<?xml version='1.0' encoding='UTF-8'?>\n<collection>"
+                    task_sleep_now_if_required(can_stop_too=True)
+
+            MARC_RECS_STR += "</collection>"
+            timestamp = time.strftime("%Y-%m-%dT%H:%M:%SZ", time.localtime())
+            marcxmlfile = 'MARCxml_booksearch' + '_' + timestamp + '_'
+            fd, marcxmlfile = tempfile.mkstemp(dir=CFG_TMPDIR, prefix=marcxmlfile, suffix='.xml')
+            os.write(fd, MARC_RECS_STR)
+            os.close(fd)
+            write_message("Composed MARCXML saved into %s" % marcxmlfile)
+
+            #Schedule the bibupload task.
+            task_id = task_low_level_submission("bibupload", "BibCirc", "-c", marcxmlfile, "-P", "3")
```

Comment:
Yes, It's needed to ensure all the uploads are done before midnight of the day of run. This has been discussed and finalized with Sebastian.